### PR TITLE
 - add get_time_range with time and  time_coverage  variables

### PR DESCRIPTION
**Pull request by nyoudjou**
- rename previous definition to get_time_range_v1

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -3,27 +3,27 @@ channels:
...
+  - gdal
+  - matplotlib
+  - netcdf4
+  - numpy
+  - pandas
+  - pyyaml
+  - s3fs
+  - scipy
+  - xarray
+  - zarr
```
Comment (forman) [inline environment.yml]: This is not acceptable. We need definite environment with fixed package versions. What is the problem with the previous settings?

File: xcube/api/gen/rbins/iproc.py [hunk 2]
```diff
@@ -52,14 +52,50 @@ def get_reprojection_info(self, dataset: xr.Dataset) -> ReprojectionInfo:
                                 xy_crs=CRS_WKT_EPSG_4326,
                                 xy_gcp_step=1)
 
-    def get_time_range(self, dataset: xr.Dataset) -> Tuple[float, float]:
+    def get_time_range_v1(self, dataset: xr.Dataset) -> Tuple[float, float]:
```
Comment (forman) [inline xcube/api/gen/rbins/iproc.py]: What is the use of `get_time_range_v1`? If it is no longer used, please remove it.

File: xcube/api/gen/snap/iproc.py [hunk 3]
```diff
@@ -52,8 +52,13 @@ def get_reprojection_info(self, dataset: xr.Dataset) -> ReprojectionInfo:
...
     def get_time_range(self, dataset: xr.Dataset) -> Tuple[float, float]:
-        t1 = dataset.attrs.get('start_date')
-        t2 = dataset.attrs.get('stop_date') or t1
+
+        if "time_coverage_start" in dataset.attrs:
+            t1 = str(dataset.attrs["time_coverage_start"])
+            t2 = str(dataset.attrs.get("time_coverage_end", t1))
+        else:
+            t1 = dataset.attrs.get('start_date')
+            t2 = dataset.attrs.get('stop_date') or t1
```
Comment (forman) [inline xcube/api/gen/snap/iproc.py]:
```suggestion
            t2 = dataset.attrs.get('stop_date', t1)
```

File: xcube/api/gen/rbins/iproc.py [hunk 4]
```diff
@@ -52,14 +52,50 @@ def get_reprojection_info(self, dataset: xr.Dataset) -> ReprojectionInfo:
...
         date = dataset.attrs.get('DATE')
         if date is None:
             raise ValueError('illegal L2 input: missing DATE attribute')
         time = dataset.attrs.get('TIME', '1200')
         days_since_1970 = get_time_in_days_since_1970(date + time)
         return days_since_1970, days_since_1970
 
+    def get_time_range(self, dataset: xr.Dataset) -> Tuple[float, float]:
```
Comment (forman) [inline xcube/api/gen/rbins/iproc.py]: This looke very much like the implementation in `DefaultInputProcessor.get_time_range()`.
Try deriving your class from `DefaultInputProcessor` so we don't duplicate code and only need a single test:

```
class RbinsSeviriHighrocSceneInputProcessor(DefaultInputProcessor):
    ...
```

**Comment by forman** [comment]
Also, please note that your branch still created failing builds, see logs, unit tests are still failing. Please fix that too.

File: xcube/api/gen/rbins/iproc.py (hunk 2 above)
Comment (nyoudjou) [reply xcube/api/gen/rbins/iproc.py]: yes, is not longer used, I will remove it

File: xcube/api/gen/snap/iproc.py (hunk 3 above)
Comment (nyoudjou) [reply xcube/api/gen/snap/iproc.py]: I am learning python3 :)

File: environment.yml (hunk 1 above)
Comment (nyoudjou) [reply environment.yml]: I had some issue with the install cube on mac os Mojave, I will try to reproduce that

**Pull request merged**